app/controllers/volunteers_controller.rb

```diff
@@ -20,32 +20,38 @@ def new
           redirect_to volunteer_path(current_user.volunteer.id)
     end
   end
-  
+
   def show
     @volunteer = current_user.volunteer
   end
-  
+
   def create
     # @repair = Repair.all
     @volunteer = Volunteer.new(volunteer_params)
     @volunteer.user_id = current_user.id
     if @volunteer.save
-          redirect_to volunteer_path(@volunteer.id)
+      flash[:success] = "Volunteer created!"
+      redirect_to volunteer_path(@volunteer)
     else
       render 'new'
     end
   end
 
   def edit
     # @repair = Repair.all
-    @volunteer = Volunteer.find(current_user.id)
+    @volunteer = Volunteer.find(params[:id])
```

Comment:
Hm, there is a problem here.  The advantage of Volunteer.find(current_user.id) is that it prevents a malicious user from editing someone else's profile.  There should be a check that params[:id] is the same as current_user.id, or something like that.